src/service/xhr-impl.js

```diff
@@ -451,37 +466,72 @@ export function assertSuccess(response) {
   });
 }
 
-
 /**
- * Response object in the Fetch API.
+ * Response object in the Fetch API. This interface is implemented by native
+ * `Response` and by the polyfill and mock response types.
  *
  * See https://developer.mozilla.org/en-US/docs/Web/API/GlobalFetch/fetch
+ *
+ * @record
```

Comment:
I would just leave this as it was, and do proper unit testing of the mock implementation.